articles/language/file-structure.md

```diff
@@ -385,3 +385,32 @@ function DotProduct(a : Double[], b : Double[]) : Double {
     return accum;
 }
 ```
+
+
+## Internal Declarations
+
+User-defined types, operations, and functions can also be declared as *internal*.
+This means that they can only be accessed from within the Q# project that they are declared in.
+When a project is used as a reference, all of its *public* (non-internal) declarations are made available, but trying to use an internal declaration from another project will produce an error.
+Internal declarations are useful for writing modular code that can be reused by other parts of your project, but still be changed later without breaking other projects that might depend on it.
+
+> [!WARNING]
+> Internal user-defined types cannot be used in the signatures of public callables or in the underlying types of public user-defined types.
```

Comment:
```suggestion
> Internal user-defined types cannot be used in the signatures of public callables, or in the underlying types of public user-defined types.
```

Reply:
Committed in #765 (but then I also ended up removing this sentence).